pandas/tools/tests/test_merge_asof.py

```diff
@@ -0,0 +1,306 @@
+        quotes = self.quotes
+
+        result = merge_asof(trades, quotes,
+                            on='time',
+                            by='ticker')
+        assert_frame_equal(result, expected)
+
+    def test_basic_categorical(self):
+
+        expected = self.asof
+        trades = self.trades.copy()
+        trades.ticker = trades.ticker.astype('category')
+        quotes = self.quotes.copy()
+        quotes.ticker = quotes.ticker.astype('category')
+
+        result = merge_asof(trades, quotes,
+                            on='time',
+                            by='ticker')
+        assert_frame_equal(result, expected)
+
+    def test_missing_right_by(self):
+
+        expected = self.asof
+        trades = self.trades
+        quotes = self.quotes
+
+        q = quotes[quotes.ticker != 'MSFT']
+        result = merge_asof(trades, q,
+                            on='time',
+                            by='ticker')
+        expected.loc[expected.ticker == 'MSFT', ['bid', 'ask']] = np.nan
+        assert_frame_equal(result, expected)
+
+    def test_basic2(self):
+
+        expected = self.read_data('asof2.csv')
+        trades = self.read_data('trades2.csv')
+        quotes = self.read_data('quotes2.csv', dedupe=True)
+
+        result = merge_asof(trades, quotes,
+                            on='time',
+                            by='ticker')
+        assert_frame_equal(result, expected)
+
+    def test_basic_no_by(self):
+        f = lambda x: x[x.ticker == 'MSFT'].drop('ticker', axis=1) \
+            .reset_index(drop=True)
+
+        # just use a single ticker
+        expected = f(self.asof)
+        trades = f(self.trades)
+        quotes = f(self.quotes)
+
+        result = merge_asof(trades, quotes,
+                            on='time')
+        assert_frame_equal(result, expected)
+
+    def test_valid_join_keys(self):
+
+        trades = self.trades
+        quotes = self.quotes
+
+        with self.assertRaises(MergeError):
+            merge_asof(trades, quotes,
+                       left_on='time',
+                       right_on='bid',
+                       by='ticker')
+
+        with self.assertRaises(MergeError):
+            merge_asof(trades, quotes,
+                       on=['time', 'ticker'],
+                       by='ticker')
+
+    def test_with_duplicates(self):
+
+        q = pd.concat([self.quotes, self.quotes],
+                      ignore_index=True).sort_values('time')
+        result = merge_asof(self.trades, q,
+                            on='time',
+                            by='ticker')
+        expected = self.read_data('asof.csv')
+        assert_frame_equal(result, expected)
+
+        result = merge_asof(self.trades, q,
+                            on='time',
+                            by='ticker',
+                            check_duplicates=False)
+        expected = self.read_data('asof.csv')
+        self.assertEqual(len(result), len(expected))
```

Comment:
maybe test here the actual dataframe result as welk? (as it should give the same result?) As this is the only place where this is tested I think?